File: testers/avro/scala-json/generated-and-checked-in/com/example/events/Address.scala
package com.example.events

import com.fasterxml.jackson.annotation.JsonProperty

/** A physical address */
case class Address(
  /** Street address */
  @JsonProperty("street") street: String,
  /** City name */
  @JsonProperty("city") city: String,
  /** Postal/ZIP code */
  @JsonProperty("postalCode") postalCode: String,
  /** Country code (ISO 3166-1 alpha-2) */
  @JsonProperty("country") country: String
)

File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/Address.java
package com.example.events;

import com.fasterxml.jackson.annotation.JsonProperty;

/** A physical address */
public record Address(
    /** Street address */
    @JsonProperty("street") String street,
    /** City name */
    @JsonProperty("city") String city,
    /** Postal/ZIP code */
    @JsonProperty("postalCode") String postalCode,
    /** Country code (ISO 3166-1 alpha-2) */
    @JsonProperty("country") String country) {
  /** Street address */
  public Address withStreet(String street) {
    return new Address(street, city, postalCode, country);
  }

  /** City name */
  public Address withCity(String city) {
    return new Address(street, city, postalCode, country);
  }

  /** Postal/ZIP code */
  public Address withPostalCode(String postalCode) {
    return new Address(street, city, postalCode, country);
  }

  /** Country code (ISO 3166-1 alpha-2) */
  public Address withCountry(String country) {
    return new Address(street, city, postalCode, country);
  }
}
